### PR TITLE
Migrate account feature from XML to Jetpack Compose

### LINKED ISSUE #173: Migrate Account feature from XML to Jetpack Compose
Currently feature Account in `presentation/account` package is implemented with XML view system. Let's convert it to Jetpack Compose. See `shop` feature for reference

### PR DESCRIPTION
**Pull request by Devendra34**
## Onboarding in Jetpack Compose (Fixes #173)
- [x] Implemented AccountScreen Composable
- [x] Deleted fragment_account.xml file and its references
- [x] Fixed PrimaryButton container color from colorPrimaryDark to colorPrimary

## Composable tree
- `AccountScreenView` (Scaffold->Column)
  - `AccountHeaderView` (Top Green colored layout)
  - `AccountContentView` (SignIn/SignOut buttons and other settings options)

## Screenshot

| Signed In State     | Signed Out State |
:-------------------------:|:-------------------------:
Screenshot 2023-10-02 at 12 21 10 AM  |  Screenshot 2023-10-01 at 11 32 56 PM

### CODE REVIEW

File: app/src/main/java/com/hieuwu/groceriesstore/presentation/utils/DensityUtils.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import androidx.annotation.DimenRes
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.unit.TextUnit
+
+@Composable
+@ReadOnlyComposable
+fun textUnit(@DimenRes id: Int): TextUnit {
```
[the remark this answers is not in the document]
Comment (Devendra34) [reply app/src/main/java/com/hieuwu/groceriesstore/presentation/utils/DensityUtils.kt:12]: The responsibility of this function is to take any dimen resource id as an input and return it corresponding TextUnit value which can be passed as `fontSize` property to any `Text` composable.

1. We use [dimensionResource API](https://developer.android.com/jetpack/compose/resources#dimensions) to get dimensions from a resource XML file.
2. The result of the 1st step is in [Dp](https://developer.android.com/reference/kotlin/androidx/compose/ui/unit/Dp#Dp(kotlin.Float)).
3. Then we convert this [Dp](https://developer.android.com/reference/kotlin/androidx/compose/ui/unit/Dp#Dp(kotlin.Float)) result to [TextUnit](https://developer.android.com/reference/kotlin/androidx/compose/ui/unit/TextUnit#value()) using [toSp](https://developer.android.com/reference/kotlin/androidx/compose/ui/unit/Density#(kotlin.Float).toSp()) extension provided on [Dp](https://developer.android.com/reference/kotlin/androidx/compose/ui/unit/Dp#Dp(kotlin.Float)) in [Density.kt](https://developer.android.com/reference/kotlin/androidx/compose/ui/unit/Density).
4. Finally return the converted [TextUnit](https://developer.android.com/reference/kotlin/androidx/compose/ui/unit/TextUnit#value())

[the remark this answers is not in the document]
Comment (hieuwu) [reply app/src/main/java/com/hieuwu/groceriesstore/presentation/utils/DensityUtils.kt:12]: Makes sense. Great work Deve 💯

[the remark this answers is not in the document]
Comment (Devendra34) [reply app/src/main/java/com/hieuwu/groceriesstore/presentation/utils/DensityUtils.kt:12]: Would you like me to put the same above comment as code document for the given method?

**Review by hieuwu — APPROVED** [review]
Lgtm

**Comment by hieuwu** [comment]
Hi @Devendra34 . Just a question. Did you create account on Supabase?

**Comment by Devendra34** [comment]
> Hi @Devendra34 . Just a question. Did you create account on Supabase?

Yes @hieuwu.

**Pull request merged**